aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -219,14 +286,15 @@ private Completable notifyContentAvailable() {
     @Nullable
     @Override
     public PendingMutation<? extends Model> peek() {
-        return mutationQueue.peek();
+        load().blockingAwait();
```

Comment:
This is a problem. We are re-loading the entire table on every `peek` - during a `drainMutationOutbox` this would be extremely inefficient. Let's think of a better approach here.

Reply:
Modified `load()` to only load one record.